pkg/controller/cloud/node_controller.go

```diff
@@ -229,21 +229,16 @@ func (cnc *CloudNodeController) updateNodeAddress(ctx context.Context, node *v1.
 		return
 	}
 	// Node that isn't present according to the cloud provider shouldn't have its address updated
-	exists, err := ensureNodeExistsByProviderID(ctx, instances, node)
+	instanceMeta, err := instances.InstanceMetadataByProviderID(ctx, node.Spec.ProviderID)
```

Comment:
The Instances interface is pretty bloated IMO. Wondering if it would be worthwhile to add an `InstancesV2` interface that is just `InstanceMetadataByProviderID` for now. If a cloud provider "implements" the v2 interface, we use the new code path here. Otherwise fallback to the existing one. Thoughts? cc @cheftako

Reply:
Started a discussion on the mailing list for this https://groups.google.com/d/msg/kubernetes-sig-cloud-provider/IYbzlMWvndQ/VgXxgGIFAQAJ

Reply:
Thanks, I have seen it.

Reply:
Hi @andrewsykim , how's it going? If we are not ready to deprecate NodeAddressesByProviderID, InstanceTypeByProviderID and InstanceExistsByProviderID, would [#90894](https://github.com/kubernetes/kubernetes/pull/90894) be reverted?

Reply:
I didn't initially realize the full intention here - but basically what you meant is that the existing implementation will cover both the old and the new interface. Which is cool.

And eventually, we will (hopefully) remove the old interface (and together with that the methods that are no longer needed).

>  I don't think we need to revert #90894, we can just do the change here.

@andrewsykim - the question then is that if we don't want to revert the part of that original PR adding the new method to the interface (and leave only the implementation addions).
We probably should, right?

Reply:
> @andrewsykim - the question then is that if we don't want to revert the part of that original PR adding the new method to the interface (and leave only the implementation addions).
We probably should, right?

Yeah sorry I wasn't clear. We should revert the addition of `InstanceMetadataByProviderID` and the deprecation of the other methods in `Instances`. I just meant we didn't have to revert the PR, we can just revert the changes here. Either way works for me :).